### DISCUSSION TITLE
Renovate with yarn splits package on upgrade

### DISCUSSION THREAD

**Discussion by Tofandel** [renovatebot/renovate#23219] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.159.7 (https://gitlab.com/renovate-bot/renovate-runner/-/blob/main/templates/renovate.gitlab-ci.yml)

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

![image](https://github.com/renovatebot/renovate/assets/6115458/51e1691c-bba5-4f0e-930b-25640f7b4933)

Notice how `@babel/core` which was in one dependency gets split into 2 dependencies

Same for subdependencies
![image](https://github.com/renovatebot/renovate/assets/6115458/d1ee9374-a050-4794-953a-7273cf897d38)

If I run `yarn upgrade` locally the depencies gets merged back together

This effectively creates problem as well as unnecessary wasted space because 2 versions of a dependency exists which causes unexpected bugs

This is also a problem that accumulates, after 4-5 upgrades I will sometimes have 5 times the subdependencies listed in the yarn lock with 5 different versions even though only one is needed

When using `npm` this problem never happens

Here is a reproduction with the renovate bot having the same output
https://github.com/Tofandel/renovate_yarn_bug/pull/2/files

### Relevant debug logs

When running `yarn upgrade @babel/core --latest` only changing the pinned dependency, I can reproduce the same lock file locally, so it's a problem with yarn itself and it's usage

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
Please look into whether the yarn dedupe options work for you

**Reply by Tofandel** [reply]
yarn dedupe
yarn dedupe v1.22.19
error The dedupe command isn't necessary. `yarn install` will already dedupe.

What a troll.. 🙄

**Reply by Tofandel** [reply]
The https://www.npmjs.com/package/yarn-deduplicate works though, and I see there is an option `yarnDedupeHighest` `yarnDedupeFewer` so probably if I enable the option locally it will work

**Reply by Tofandel** [reply, marked as answer]
Adding this to the renovate config does the trick
```json
  "postUpdateOptions": ["npmDedupe","yarnDedupeFewer"]
```

Just think it should be enabled by default
